bin/build-mb/test/build/licenses_test.clj

```diff
@@ -0,0 +1,215 @@
+(ns build.licenses-test
+  (:require [build.licenses :as lic]
+            [clojure.data.xml :as xml]
+            [clojure.edn :as edn]
+            [clojure.java.io :as io]
+            [clojure.string :as str]
+            [clojure.test :refer [deftest is run-tests testing]]
+            [metabuild-common.core :as u])
+  (:import (java.io StringReader StringWriter)
+           java.util.jar.JarFile))
+
+(def classpath-urls (str/split (System/getProperty "java.class.path")
+                               (re-pattern lic/classpath-separator)))
+
+(defn jar-filename-from-cp
+  [jar-filename classpath-urls]
+  (let [[x & rst :as found] (filter #(when (str/includes? % jar-filename) %) classpath-urls)]
+    (cond (seq rst) (throw (ex-info (str "Multiple jars found for " jar-filename)
+                                    {:filename jar-filename
+                                     :found    found}))
+          (not x)   (throw (ex-info (str "No results found for " jar-filename)
+                                    {:filename jar-filename}))
+          x         x)))
+
+(defn parse [rdr] (xml/parse rdr :skip-whitespace true))
+
+(def clojure-xml "<?xml version=\"1.0\" encoding=\"UTF-8\"?>
+<project xmlns=\"http://maven.apache.org/POM/4.0.0\" xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" xsi:schemaLocation=\"http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd\">
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>org.clojure</groupId>
+  <artifactId>clojure</artifactId>
+  <name>clojure</name>
+  <packaging>jar</packaging>
+  <version>1.10.3</version>
+
+  <licenses>
+    <license>
+      <name>Eclipse Public License 1.0</name>
+      <url>http://opensource.org/licenses/eclipse-1.0.php</url>
+      <distribution>repo</distribution>
+    </license>
+  </licenses>
+</project>"
+  )
```

Comment:
Would it be easier to stick these in separate `.xml` files that we slurp as opposed to embedding in the source code like this?

Reply:
Up to you. this is about 20 lines of pretty readable text, and it is inline so it's easy to have a concrete notion of what is going on. I can move it to knock down the clutter but it kinda feels less easy to me. I think your judgement on our test style should rule here so give me the final answer. I'll do either.